src/definitions/mso.rs

```diff
@@ -1,22 +1,67 @@
+//! This module contains the definitions for the `MSO` (Mobile Security Object) structure.
+//!
+//! The `MSO structure represents a mobile security object, which is used in cryptographic operations
+//! within the system. It contains information such as the version, digest algorithm, value digests,
+//! device key info, document type, and validity info.
+//!
+//! # Examples
+//!
+//! ```ignore
+//! use spruceid::definitions::mso::{Mso, DigestAlgorithm};
+//! use std::collections::BTreeMap;
+//!
+//! // Create a new MSO object
+//! let mso = Mso {
+//!     version: String::from("1.0"),
+//!     digest_algorithm: DigestAlgorithm::SHA256,
+//!     value_digests: BTreeMap::new(),
+//!     device_key_info: Default::default(),
+//!     doc_type: String::from("document"),
+//!     validity_info: Default::default(),
+//! };
+//!
+//! // Print the MSO object
+//! println!("{:?}", mso);
+//! ```
+//!
+//! # Notes
+//!
+//! - [DigestId] struct represents an unsigned integer between `0` and `(2^31 - 1)` inclusive.  
+//!   It is enforced to be positive.
+//! - [DigestIds] type is a [BTreeMap] that maps [DigestId] to [ByteStr].
+//! - [DigestAlgorithm] enum represents different digest algorithms, such as `SHA-256, `SHA-384,
+//!   and `SHA-512`.
 use crate::definitions::{helpers::ByteStr, DeviceKeyInfo, ValidityInfo};
 use serde::{Deserialize, Serialize};
 use std::collections::BTreeMap;
 
-/// DigestId is a unsigned integer between 0 and (2^31 - 1) inclusive.
+/// DigestId is a unsigned integer between `0` and `(2^31 - 1)` inclusive.
 /// Therefore the most straightforward way to represent it is as a i32 that is enforced to be
 /// positive.
 #[derive(Clone, Debug, Serialize, Deserialize, Eq, Ord, PartialEq, PartialOrd, Copy, Hash)]
 pub struct DigestId(i32);
 pub type DigestIds = BTreeMap<DigestId, ByteStr>;
 
+/// Represents an [Mso] object.
 #[derive(Clone, Debug, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct Mso {
```

Comment:
I still think the `Mso` struct could benefit from a full name, `MobileSecurityObject`.

Reply:
It's used prolifically as `MSO` throughout the spec, so perhaps keeping the code aligned is a good idea.

Reply:
Make sense and avoids a serde renaming. Disregard my comment on struct renaming.